app/controllers/chargeback_controller.rb

```diff
@@ -205,25 +205,22 @@ def cb_rates_delete
     if !params[:id] # showing a list
       rates = find_checked_items
       if rates.empty?
-        render_flash(_("No Chargeback Rate were selected for deletion"), :error)
+        add_flash(_("No Chargeback Rates were selected for deletion"), :error)
       end
     else # showing 1 rate, delete it
-      cb_rate = ChargebackRate.find_by_id(params[:id])
+      cb_rate = ChargebackRate.find_by(:id => params[:id])
+      self.x_node = x_node.split('_').first
       if cb_rate.nil?
-        render_flash(_("Chargeback Rate no longer exists"), :error)
+        add_flash(_("Chargeback Rate no longer exists"), :error)
       else
         rates.push(params[:id])
-        self.x_node = "xx-#{cb_rate.rate_type}"
       end
     end
-    process_cb_rates(rates, 'destroy') unless rates.empty?
-    if flash_errors?
-      javascript_flash
-    else
-      cb_rates_list
-      @right_cell_text = _("%{typ} Chargeback Rate") % {:typ => x_node.split('-').last}
-      replace_right_cell(:replace_trees => [:cb_rates])
-    end
+    process_cb_rates(rates, 'destroy') if rates.present?
+
+    cb_rates_list
+    @right_cell_text = _("%<typ>s Chargeback Rates") % {:typ => x_node.split('-').last}
+    replace_right_cell(:replace_trees => [:cb_rates])
```

Comment:
@hstastna i have not tested this in UI but wondering why this was moved out of if/else block, i don't think we need to call `cb_rates_list` and replace_right_cell methods if there were flash errors such as "no records selected for deletion" in that case we should simply render flash message partial.

Reply:
As I remember, this was the best option to fix another issues (and to not break anything else) which came with fixing the bug: I came to the situation when I needed to render flash message if some error occurs AND to display remaining rates properly (which were not deleted or which shouldn't  have been deleted). Because if we check more than one rate in the list and try to delete them, some of them are successfully deleted and some of them not (like Default rates) and we need to show error message AND the actual list of the rates. We need to update the list after deleting. It was not very good that the if/else block came after deleting the rates.